### ISSUE TITLE
Debug specific test filter not applied on Windows

### ISSUE THREAD

**Issue by a-type** [vitest-dev/vscode#64]
**Describe the bug**
When trying to debug a specific test block, the debug process just runs all tests in the file on Windows with Yarn.

**To Reproduce**
Steps to reproduce the behavior on [the example project](https://github.com/zxch3n/vitest-ext-basic-example):

1. `yarn install`
2. Add a second test block
```
import { describe, it, expect} from 'vitest';

describe('add', () => {
  it ('1 + 1', () => {
    expect(1 + 1).toBe(2);
  })

  it ('2 + 2', () => {
    expect(2 + 2).toBe(4);
  })
})
```
3. Add a breakpoint on line 5, inside the "1 + 1" test
4. Open the Vitest sidebar and select "Debug test" on "2 + 2"
5. The breakpoint is hit, indicating "1 + 1" is being run

**Expected behavior**
When debugging a specific test, I expect to only run that test

**Screenshots**
![image](https://user-images.githubusercontent.com/2829772/180664850-1686efb4-a2de-4cd9-afc6-d76f065b65a0.png)
Shown above: "2 + 2" is being debugged, but the debugger has broken inside "1 + 1"

The full output for the debug test run:

```
"C:\Program Files\nodejs\node.exe" .\node_modules\vitest\vitest.mjs test D:/vitest-ext-basic-example/add.test.ts -t ""add 2 + 2"" --api 52918
 DEV  v0.12.10 d:/vitest-ext-basic-example
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10480
      API started at http://localhost:52918
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10480
Could not read source map for file:///d:/vitest-ext-basic-example/@vite/env: ENOENT: no such file or directory, open 'd:\vitest-ext-basic-example\@vite\env.mjs.map'
```

**Environment**

(Paste info.txt content generated by [the example project](https://github.com/zxch3n/vitest-ext-basic-example))

- OS: [e.g. macOS] Windows 11
- VSCode version: 1.69.2
- Vitest version: 0.12.10
- Vitest plugin version: v0.2.23 (prerelease)

(filled manually, the setup script does not work on Windows either)

```
Vitest settings updated
'C:\Users\a-typ\AppData\Local\Programs\Microsoft' is not recognized as an internal or external command,
operable program or batch file.
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

**Comment by lmexo** [comment]
This is probably related to the quoting of the test name pattern on windows devices: It is quoted and then later passed to the debug config, which quotes the argument againt, thus, on windows, the comandline looks like: ` ... tests/add.test.ts -t ""multi word named task""`. What windows effectively sees is ` ... tests/add.test.ts -t multi word named task`, so the pattern is multi, and "word", "named" and "task" is passed to the filter arguments.

For what I tested, the bug can simply be fixed by removing the following:
https://github.com/vitest-dev/vscode/blob/dd70a041b8d955ed821b53684104a1f630044ea0/src/pure/runner.ts#L82-L84

**Comment by zxch3n** [comment]
@a-type does it work now on pre-release version?

**Comment by a-type** [comment]
Tests no longer run at all with any run mode with the latest version:

```
Test result not found. 
Are there tests with the same name?
Can you run vitest successfully on this file? Does it need custom option to run?

Vitest output:
'.' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by a-type** [comment]
Debug run now states `No test files found, exiting with code 1`.  Full output:

```
"C:\Program Files\nodejs\node.exe" .\node_modules\vitest\vitest.mjs D:/vitest-ext-basic-example/add.test.ts -t "add 2 + 2" --api 60210
 DEV  v0.12.10 d:/vitest-ext-basic-example
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10480
      API started at http://localhost:60210
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10480
filter:  D:/vitest-ext-basic-example/add.test.ts
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10365
include: **/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10367
ignore:  /\/node_modules\//, /\/dist\//
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10369

No test files found, exiting with code 1
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10483
Process exited with code 1
```

Perhaps because of the forward slashes in the test file path? It's unclear, considering I can run `node .\node_modules\vitest\vitest.mjs D:/vitest-ext-basic-example/add.test.ts -t "add 2 + 2" --api 60210` in PowerShell and it works fine.

**Comment by a-type** [comment]
It might actually be the `+` symbol in the filter. Running `-t "add 2"` correctly filters to `2 + 2`, but `-t add 2 + 2` skips everything.

**Comment by zxch3n** [comment]
@a-type thank you for your info. It's weird we need to escape `+`. Can you run tests correctly on this version?

**Comment by a-type** [comment]
My reproduction case still does not work. Attempting to debug an individual test in the example repro does not run anything.

This does not appear to be related to `+` after all (even though that does seem to be a problem). Apologies for the distraction. For example:

```
import { describe, it, expect} from 'vitest';

describe('add', () => {
  it ('1 + 1', () => {
    expect(1 + 1).toBe(2);
  })

  it ('2 + 2', () => {
    expect(2 + 2).toBe(4);
  })

  it('foo', () => {
    expect('foo').toBe('foo');
  })
})
```

Running just `add foo`:
![image](https://user-images.githubusercontent.com/2829772/183302179-f89cd0e0-818f-498a-840e-93baacfef143.png)

Outputs:
![image](https://user-images.githubusercontent.com/2829772/183302185-ed90f9e7-0a01-43a0-9139-bd33686589fa.png)

```
"C:\Program Files\nodejs\node.exe" .\node_modules\vitest\vitest.mjs D:/vitest-ext-basic-example/add.test.ts -t "add foo" --api 50098
 DEV  v0.12.10 d:/vitest-ext-basic-example
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10480
      API started at http://localhost:50098
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10480
filter:  D:/vitest-ext-basic-example/add.test.ts
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10365
include: **/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10367
ignore:  /\/node_modules\//, /\/dist\//
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10369

No test files found, exiting with code 1
node_modules/vitest/dist/chunk-vite-node-externalize.464ab3dd.js:10483
Process exited with code 1
```

**Comment by jtamyrc** [comment]
Would this issue related to this issue #65 ?

The drive name is a capital case, e.g. `C:/` or `D:/`

**Comment by a-type** [comment]
That could be it, yeah.

**Comment by IGx89** [comment]
FYI, the commit made to address this issue (https://github.com/vitest-dev/vscode/commit/a6019a45d45abb0411f3e6a67eaf5c5f00dead2e) caused quotes to be set properly when debugging (which uses a VS Code API) but turned off quotes when running (which uses spawn), effectively undoing part of #24.

Run commandline:
`"C:\Program Files\nodejs\node.exe" .\node_modules\vitest\vitest.mjs /xyz/test/unit/vue-component/vueComponent.spec.ts -t "Vue Component init" --api.port 63875 --api.host 127.0.0.1`

Debug commandline:
`c:/xyz/node_modules/.bin/vitest.cmd /xyz/test/unit/vue-component/vueComponent.spec.ts -t Vue Component init`

**Comment by IGx89** [comment]
I've found a more targeted fix that I'll be PR'ing shortly. As for the issue reported here, I believe it was fully fixed by #94 and think this could be closed now.
